homeassistant/components/adax/config_flow.py

```diff
@@ -5,69 +5,138 @@
 
-from .const import ACCOUNT_ID, DOMAIN
-
-_LOGGER = logging.getLogger(__name__)
-
-STEP_USER_DATA_SCHEMA = vol.Schema(
-    {vol.Required(ACCOUNT_ID): int, vol.Required(CONF_PASSWORD): str}
+from .const import (
+    ACCOUNT_ID,
+    CLOUD,
+    CONNECTION_TYPE,
+    DOMAIN,
+    LOCAL,
+    WIFI_PSWD,
+    WIFI_SSID,
 )
 
-
-async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
-    """Validate the user input allows us to connect."""
-    account_id = data[ACCOUNT_ID]
-    password = data[CONF_PASSWORD].replace(" ", "")
-
-    token = await adax.get_adax_token(
-        async_get_clientsession(hass), account_id, password
-    )
-    if token is None:
-        _LOGGER.info("Adax: Failed to login to retrieve token")
-        raise CannotConnect
+_LOGGER = logging.getLogger(__name__)
 
 
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Adax."""
 
     VERSION = 1
 
-    async def async_step_user(
-        self, user_input: dict[str, Any] | None = None
-    ) -> FlowResult:
+    async def async_step_user(self, user_input=None):
         """Handle the initial step."""
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONNECTION_TYPE, default=CLOUD): vol.In(
+                    (
+                        CLOUD,
+                        LOCAL,
+                    )
+                )
+            }
+        )
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=data_schema,
+            )
+
+        if user_input[CONNECTION_TYPE] == LOCAL:
+            return await self.async_step_local()
+        return await self.async_step_cloud()
+
+    async def async_step_local(self, user_input=None):
+        """Handle the local step."""
+        data_schema = vol.Schema(
+            {vol.Required(WIFI_SSID): str, vol.Required(WIFI_PSWD): str}
+        )
         if user_input is None:
             return self.async_show_form(
-                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+                step_id="local",
+                data_schema=data_schema,
             )
 
+        wifi_ssid = user_input[WIFI_SSID].replace(" ", "")
+        wifi_pswd = user_input[WIFI_PSWD].replace(" ", "")
+        configurator = adax_local.AdaxConfig(wifi_ssid, wifi_pswd)
+
+        try:
+            if not await configurator.configure_device():
+                return self.async_show_form(
+                    step_id="local",
+                    data_schema=data_schema,
+                    errors={"base": "cannot_connect"},
+                )
+        except adax_local.HeaterNotAvailable:
+            return self.async_abort(reason="heater_not_available")
+        except adax_local.HeaterNotFound:
+            return self.async_abort(reason="heater_not_found")
+        except adax_local.InvalidWifiCred:
+            return self.async_abort(reason="invalid_wifi_cred")
```

Comment:
Usually we give users a chance to try again and present the `invalid_auth` error (aka `[%key:common::config_flow::error::invalid_auth%]`)